includes/wizards/class-donations-wizard.php

```diff
@@ -0,0 +1,380 @@
+			return new WP_Error(
+				'newspack_missing_required_plugin',
+				esc_html__( 'The required plugins are not installed and activated. Install and/or activate them to access this feature.', 'newspack' ),
+				[
+					'status' => 400,
+					'level'  => 'fatal',
+				]
+			);
+		}
+		return true;
+	}
+
+	/**
+	 * API endpoint for getting donation settings.
+	 *
+	 * @return WP_REST_Response containing info.
+	 */
+	public function api_get_donation_settings() {
+		$required_plugins_installed = $this->check_required_plugins_installed();
+		if ( is_wp_error( $required_plugins_installed ) ) {
+			return rest_ensure_response( $required_plugins_installed );
+		}
+
+		return rest_ensure_response( self::get_donation_settings() );
+	}
+
+	/**
+	 * Get the donation settings.
+	 *
+	 * @return Array of dontion settings (See $settings at the top of the method for format).
+	 */
+	public static function get_donation_settings() {
+		$settings = [
+			'name' => '',
+			'suggestedAmount' => 15.00,
+			'image' => false,
+		];
+
+		$product_id = get_option( NEWSPACK_DONATION_PRODUCT_ID, 0 );
+		if ( ! $product_id ) {
+			return $settings;
+		}
+
+		$product = \wc_get_product( $product_id );
+		if ( ! $product || 'grouped' !== $product->get_type() ) {
+			return $settings;
+		}
+
+		$settings['name'] = $product->get_name();
+		$settings['image'] = [
+			'id'  => $product->get_image_id(),
+			'url' => $product->get_image_id() ? current( wp_get_attachment_image_src( $product->get_image_id(), 'woocommerce_thumbnail' ) ) : wc_placeholder_img_src( 'woocommerce_thumbnail' ),
+		];
+
+		$suggested_donation = $product->get_meta( NEWSPACK_DONATION_SUGGESTED_AMOUNT_META, true );
+		if ( $suggested_donation ) {
+			$settings['suggestedAmount'] = floatval( $suggested_donation );	
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * API endpoint for setting the donation settings.
+	 *
+	 * @param WP_REST_Request $request Request containing settings.
+	 * @return WP_REST_Response with the latest settings.
+	 */
+	public function api_set_donation_settings( $request ) {
+		$required_plugins_installed = $this->check_required_plugins_installed();
+		if ( is_wp_error( $required_plugins_installed ) ) {
+			return rest_ensure_response( $required_plugins_installed );
+		}
+		return rest_ensure_response( self::set_donation_settings( $request->get_params() ) );
+	}
+
+	/**
+	 * Set the donation settings.
+	 *
+	 * @param array $args Array of settings info. See $defaults at top of this method for format.
+	 * @return array Updated settings.
+	 */
+	public static function set_donation_settings( $args ) {
+		$defaults = [
+			'name' => '',
+			'suggestedAmount' => 0,
+			'image_id' => 0,
+		];
+		$args = wp_parse_args( $args, $defaults );
+
+		// Create the product if it hasn't been created yet.
+		$product_id = get_option( NEWSPACK_DONATION_PRODUCT_ID, 0 );
+		if ( ! $product_id ) {
```

Comment:
I believe your products were in 'trash' status, which is why no new products were created. If it had been fully deleted, it would have caught and rebuilt the product [in this section](https://github.com/Automattic/newspack-plugin/pull/196/files/b7495e5b8f78c9230495b90c17d2aad6153d7adb#diff-0d6c175a3c25fdff0d703a9d51d4b591R215). I've added some code to [ensure our donations products are 'publish' status](https://github.com/Automattic/newspack-plugin/pull/196/commits/04d837a6b2d0d12133579531f2e4d5b38b1c8726) which should prevent this issue in the future.